QuickFIXn/Transport/SslStreamFactory.cs

```diff
@@ -176,29 +176,43 @@ private bool VerifyRemoteCertificate(
             return false;
         }
 
+        // If CA Certificate is specified then validate against the CA certificate, otherwise it is validated against the installed certificates
+        X509Chain chain;
         if (string.IsNullOrEmpty(_socketSettings.CACertificatePath)) {
             _nonSessionLog.OnEvent("CACertificatePath is not specified");
-            return false;
-        }
-
-        string caCertPath = StringUtil.FixSlashes(_socketSettings.CACertificatePath);
+            chain = new X509Chain();
 
-        // If CA Certificate is specified then validate against the CA certificate, otherwise it is validated against the installed certificates
-        X509Certificate2? cert = SslCertCache.LoadCertificate(caCertPath, null);
-        if (cert is null) {
-            _nonSessionLog.OnEvent(
-                $"Certificate '{caCertPath}' could not be loaded from store or path '{Directory.GetCurrentDirectory()}'");
-            return false;
+            // Set the chain policy
+            if (_socketSettings.CheckCertificateRevocation) {
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.Online;
+            }
+            else {
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+            }
+            chain.ChainPolicy.RevocationFlag = X509RevocationFlag.EntireChain;
+            chain.ChainPolicy.VerificationFlags = X509VerificationFlags.NoFlag;
+            chain.ChainPolicy.VerificationTime = DateTime.Now;
         }
+        else
+        {
+            string caCertPath = StringUtil.FixSlashes(_socketSettings.CACertificatePath);
+            
+            X509Certificate2? cert = SslCertCache.LoadCertificate(caCertPath, null);
+            if (cert is null) {
+                _nonSessionLog.OnEvent(
+                    $"Certificate '{caCertPath}' could not be loaded from store or path '{Directory.GetCurrentDirectory()}'");
+                return false;
+            }
 
-        X509Chain chain0 = new X509Chain();
-        chain0.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
-        // add all your extra certificate chain
+            chain = new X509Chain();
+            chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+            // add all your extra certificate chain
 
-        chain0.ChainPolicy.ExtraStore.Add(cert);
-        chain0.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
-        bool isValid = chain0.Build((X509Certificate2)certificate);
+            chain.ChainPolicy.ExtraStore.Add(cert);
```

Comment:
If you provide a certificate file, it seems to bypass all checks on the certificate entirely, albeit, in a very verbose way.